pkg/ottl/ottlfuncs/func_delete_key.go

```diff
@@ -22,9 +22,26 @@ import (
 	"github.com/open-telemetry/opentelemetry-collector-contrib/pkg/ottl"
 )
 
-func DeleteKey[K any](target ottl.Getter[K], key string) (ottl.ExprFunc[K], error) {
+type DeleteKeyArguments[K any] struct {
+	Target ottl.Getter[K] `ottlarg:"0"`
+	Key    string         `ottlarg:"1"`
+}
+
+type DeleteKeyFactory[K any] struct{}
+
+func (f DeleteKeyFactory[K]) Name() string { return "delete_key" }
+
+func (f DeleteKeyFactory[K]) CreateEmptyArguments() ottl.Arguments {
+	return &DeleteKeyArguments[K]{}
+}
+
+func (f DeleteKeyFactory[K]) CreateFunction(ctx context.Context, fCtx ottl.FunctionContext, args ottl.Arguments) (ottl.ExprFunc[K], error) {
+	return deleteKey(ctx, fCtx, args.(*DeleteKeyArguments[K])), nil
```

Comment:
> args.(*DeleteKeyArguments[K])

Are we okay making this type conversion without checking the result? I feel like we can be confident about the result (and checking this would be verbose), but technically we are risking a panic.

Reply:
We can't risk a panic :/

Reply:
I'm good with that, I've added a check.